soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -117,46 +117,102 @@ BUILD_ASSERT(XCHAL_EXCM_LEVEL == 5);
 
 int cavs_idc_smp_init(const struct device *dev);
 
+#define CxL1CCAP (*(volatile uint32_t *)0x9F080080)
+#define CxL1CCFG (*(volatile uint32_t *)0x9F080084)
+#define CxL1PCFG (*(volatile uint32_t *)0x9F080088)
+
+/* "Data/Instruction Cache Memory Way Count" fields */
+#define CxL1CCAP_DCMWC ((CxL1CCAP >> 16) & 7)
+#define CxL1CCAP_ICMWC ((CxL1CCAP >> 20) & 7)
+
+static ALWAYS_INLINE void enable_l1_cache(void)
+{
+	uint32_t reg;
+
+#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V25
+	/* First, on cAVS 2.5 we need to power the cache SRAM banks
+	 * on!  Write a bit for each cache way in the bottom half of
+	 * the L1CCFG register and poll the top half for them to turn
+	 * on.
+	 */
+	uint32_t dmask = BIT(CxL1CCAP_DCMWC) - 1;
+	uint32_t imask = BIT(CxL1CCAP_ICMWC) - 1;
+	uint32_t waymask = (imask << 8) | dmask;
+
+	CxL1CCFG = waymask;
+	while (((CxL1CCFG >> 16) & waymask) != waymask) {
+	}
+
+	/* Prefetcher also power gates, same interface */
+	CxL1PCFG = 1;
+	while ((CxL1PCFG & 0x10000) == 0) {
+	}
```

Comment:
may need timeouts and errors here to cover any IP misconfiguration on host side or from other FWs

Reply:
Here I think there's less value.  This is REALLY early code, at this stage where the cache isn't enabled we can't communicate with other cores through memory, we can't take spinlocks, etc... So basically we can't even log an error if we detect one.